src/functor.jl

```diff
@@ -38,7 +38,7 @@ end
 
 # See https://github.com/FluxML/Functors.jl/issues/2 for a discussion regarding the need for
 # cache.
-function fmap(f, x; cache = IdDict())
+function fmap(f, x; predicate = isleaf, cache = IdDict())
   haskey(cache, x) && return cache[x]
-  cache[x] = isleaf(x) ? f(x) : fmap1(x -> fmap(f, x, cache = cache), x)
+  cache[x] = predicate(x) ? f(x) : fmap1(x -> fmap(f, x, cache = cache), x)
```

Comment:
```suggestion
  xnew = predicate(x) ? f(x) : fmap1(x -> fmap(f, x, cache = cache), x)
  cache[x] = xnew
  return xnew
```
Just a chance to make these rather obscure functions a bit more readable

Reply:
actually there is a bug, should pass over the predicate:
```suggestion
  xnew = predicate(x) ? f(x) : fmap1(x -> fmap(f, x, cache = cache, predicate = predicate), x)
  cache[x] = xnew
  return xnew
```
Better add tests catching this